### ISSUE TITLE
[BUG][Python][v7.0.0-beta] Enum models generated as BaseModel

### ISSUE THREAD

**Issue by yanatan16** [OpenAPITools/openapi-generator#16203]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

When using a Component Schema that is an enum, the corresponding model generated is not an enum type (it is a pydantic BaseModel, which is an object type) and doesn't deserialize or instantiate correctly. According to [pydantic's docs](https://docs.pydantic.dev/latest/usage/types/enums/), the way to represent an enum is using the standard library `enum.Enum`.

##### openapi-generator version

Generated with docker hub latest on 7/27/23. [Link to docker hub tag](https://hub.docker.com/layers/openapitools/openapi-generator-cli/latest/images/sha256-5ce478c02db25a7d01fcb582af6cd10bf4c550c9fbe446c08f7ccb90d73fd28d?context=explore). This is a regression compared to 6.6.0, which worked for enums.

##### OpenAPI declaration file content or url

```yaml
openapi: 3.1.0
info:
  version: 1.0.0
  title: Swagger Petstore
  license:
    name: MIT
    identifier: MIT
servers:
  - url: https://petstore3.swagger.io/api/v3
paths:
  /pets:
    get:
      summary: Get pets
      operationId: getPets

      responses:
        '200':
          description: 'Get Pets'
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: '#/components/schemas/Cat'
components:
  schemas:
    Kind:
      type: string
      enum: [Siamese, Tabby]
      title: Kind
      description: cat kind
    Cat:
      type: object
      title: Cat
      description: a cat
      required: [kind]
      properties:
        kind:
          $ref: '#/components/schemas/Kind'
```

##### Generation Details

```
docker run --rm -v ${PWD}:/local openapitools/openapi-generator-cli:latest generate -g python --additional-properties=generateSourceCodeOnly=true,packageName=client -o /local/ -i /local/openapi.yaml
```

##### Steps to reproduce

```
$ PYTHONPATH=. ipython
Python 3.9.12 (main, Apr  5 2022, 01:52:34)

In [1]: from client.models.kind import Kind

In [2]: Kind("Siamese")
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 1
----> 1 Kind("Siamese")

File /opt/homebrew/Caskroom/miniconda/base/lib/python3.9/site-packages/pydantic/main.py:333, in pydantic.main.BaseModel.__init__()

TypeError: __init__() takes exactly 1 positional argument (2 given)
```

If I regenerate with `v6.6.0`, it works:

```
$ PYTHONPATH=. ipython
Python 3.9.12 (main, Apr  5 2022, 01:52:34)

In [1]: from client.models import Kind

In [2]: Kind("Siamese")
Out[2]: 'Siamese'
```

**Comment by fa0311** [comment]
Changing the openapi version to 3.0.x solves this issues

**Comment by francipvb** [comment]
> Changing the openapi version to 3.0.x solves this issues.
> 
> ```yaml
> openapi: 3.0.0
> ...
> ```

But sometimes we can't do that.

**Comment by wing328** [comment]
@fracipvb I just merged a PR to provide beta support for 3.1.0 spec. Please given it a try with the latest master (snapshot version in project's readme)

**Comment by yanatan16** [comment]
Works for me. Thanks!
